cpp/ql/lib/semmle/code/cpp/dataflow/internal/DataFlowImplConsistency.qll

```diff
@@ -9,6 +9,19 @@ private import tainttracking1.TaintTrackingParameter::Private
 private import tainttracking1.TaintTrackingParameter::Public
 
 module Consistency {
+  private newtype TConsistencyConfiguration = MkConsistencyConfiguration()
+
+  /** A class for configuring the consistency queries. */
+  class ConsistencyConfiguration extends TConsistencyConfiguration {
+    string toString() { none() }
+
+    /** Holds if `n` should be excluded from the consistency test `postWithInFlow`. */
+    predicate postWithInFlowExclude(Node n) { none() }
```

Comment:
I'm curious about this one. In C++ we have _lots_ of flow steps into `PostUpdateNode`s due to conversions in between adjacent `storeSteps`, and I can definitely see us using this configuration to disable some of these consistency failures.

I see you've disabled this consistency check for Ruby's `SummaryNode`s. What's the motivation for this?

Reply:
It is because the node synthesis framework that underpins the shared flow summary library short-cuts things a bit, and does create flow into post-update nodes.

Reply:
I see. Thanks for the explanation!